wandb/sdk/lib/git.py

```diff
@@ -62,7 +62,11 @@ def enabled(self) -> bool:
     def root(self) -> Optional[str]:
         if not self.repo:
             return None
-        return self.repo.git.rev_parse("--show-toplevel")
+        try:
+            return self.repo.git.rev_parse("--show-toplevel")
+        except exc.GitCommandError as e:
+            logger.error(f"git root error: {e}")
```

Comment:
not a blocker
we should probably capture some telemetry for this if it is easy..

Reply:
hmm... what would we do with that information? it usually happens when the home dir of the user running the script is owned by someone else.
it's easy to do, just kind of clumsy: I'd set a flag on the GitRepo object if that happened, then would check for that in Settings as it's one of the places where it's used, and then pick it up in init. Should I still do it?